dnadoh/analyze.py

```diff
@@ -0,0 +1,169 @@
+import argparse
+
+import pandas as pd
+import matplotlib.pyplot as plt
+import plotly.express as px
+import plotly.graph_objects as go
+from plotly.subplots import make_subplots
+
+from plotly.tools import FigureFactory as FF
+from statannot import add_stat_annotation
+from dash import Dash, dcc, html, Input, Output, dash_table
+from scipy import stats
+
+from assemble import assemble
+from plotting_utils import add_pvalue_annotation
+
+BASES = ["A", "C", "G", "T"]
+
+
+def get_unique_locations(df: pd.DataFrame):
+    """Get unique locations with an alternate base"""
+    locations = df["location"].drop_duplicates().dropna().to_list()
```

Comment:
why not just `return df["location"]...etc...` ? is the temp variable so that you can debug (which is perfectly reasonable)?

Reply:
No, to be honest I've always liked assigning a variable name even to a one line returnable statement because I find it easier to understand. But that's probably why the function docstring is there!